napari/layers/base/base.py

```diff
@@ -748,6 +764,32 @@ def _reset_editable(self) -> None:
     def _on_editable_changed(self) -> None:
         """Executes side-effects on this layer related to changes of the editable state."""
 
+    @property
+    def axis_labels(self) -> tuple[str, ...]:
+        """tuple of axis labels for the layer."""
+        return self._transforms['data2physical'].axis_labels
+
+    @axis_labels.setter
+    def axis_labels(self, axis_labels: Optional[Sequence[str]]) -> None:
+        prev = self._transforms['data2physical'].axis_labels
+        # mypy bug https://github.com/python/mypy/issues/3004
+        self._transforms['data2physical'].axis_labels = axis_labels  # type: ignore[assignment]
+        if self._transforms['data2physical'].axis_labels != prev:
+            self.events.axis_labels()
+
+    @property
+    def units(self) -> tuple[pint.Unit, ...]:
+        """List of units for the layer."""
+        return self._transforms['data2physical'].units
+
+    @units.setter
+    def units(self, units: Optional[Sequence[pint.Unit]]) -> None:
+        prev = self.units
+        # mypy bug https://github.com/python/mypy/issues/3004
+        self._transforms['data2physical'].units = units  # type: ignore[assignment]
+        if self.units != prev:
+            self.events.units()
```

Comment:
I'm never really sure about whether events should include their value — I think yes? Should this be `self.events.units(value=self.units)`? (Happy to have this discussion in a follow-up issue/PR.)

Reply:
We need to first resolve the problem that we discussed here: https://github.com/pyapp-kit/psygnal/pull/296#pullrequestreview-1948842934
Otherwise, signal could contain wrong value of units (different than current one). 

It may be a part of migration to psygnal.